### PR TITLE
Made all filters work in conjunction with one another on rooms page

### PR DESCRIPTION
**Pull request by Princu7**
Partially fixes the issue #1318 

Changes: 
* Makes all different filters (starred, search and room filter) work in conjunction with one another on the rooms page.

### CODE REVIEW

File: src/backend/templates/rooms.hbs [hunk 1]
```diff
@@ -319,58 +319,54 @@
...
-                    $(this).show();
-                  }
-                  else {
-                    $(this).hide();
+          $('.date-filter').each(function() {
+            var flag = 1;
+            $(this).find('.venue-filter').each(function() {
+              var name = $(this).find('.text').text();
+              var flagVenue = 1;
+              if(roomFilterMode && (name != roomName)) {
```
Comment (aayusharora) [inline src/backend/templates/rooms.hbs]: Use `!==` here

File: src/backend/templates/rooms.hbs [hunk 2]
```diff
@@ -319,58 +319,54 @@
...
+                $(this).hide();
+                return;
+              }
+              $(this).find('.room-filter').each(function() {
+                var id = $(this).attr('id');
+                var elemFlag = 1;
+                if ($(this).find('.tip-description').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
```
Comment (aayusharora) [inline src/backend/templates/rooms.hbs]: Can you please write the `chaining` better?
`$(this).find('.tip-description').text().toUpperCase()` to      
                                                                                                    `$(this).`
                                                                                                    ` find('.tip-description').`
                                                                                                     ` text().`
                                                                                                    `  toUpperCase()`

File: src/backend/templates/rooms.hbs [hunk 3]
```diff
@@ -319,58 +319,54 @@
...
+                  elemFlag = 0;
+                  if(mode && temp[id] != 1) {
+                    elemFlag = 1;
                   }
-                });
-                if (flagVenue == 1) {
-                  $(this).hide();
-                } else {
+                }
+                if(elemFlag == 0) {
```
Comment (aayusharora) [inline src/backend/templates/rooms.hbs]: === here

**Comment by Princu7** [comment]
@aayusharora Thanks so much for the review :) I have made the appropriate changes. I am not much clear about the chaining part. There are many OR conditions there and doing it that way can be much longer. Or maybe I might have understood it wrong. Please elaborate. Thanks!

**Test Server**
http://secure-meadow-20680.herokuapp.com

**Comment by Princu7** [comment]
Here is the test website for review
http://princu7.github.io/websiteTest
@aayusharora @geekyd @sumedh123 @harshitagupta30 Please review. Thanks :)

**Pull request merged**